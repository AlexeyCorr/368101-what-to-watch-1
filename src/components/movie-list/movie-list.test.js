import React from 'react';
import renderer from 'react-test-renderer';

import MovieList from './movie-list.jsx';

const mock = {
  films: [
    {
      genre: `Romance`,
      picture: `img/bohemian-rhapsody.jpg`,
      title: `Bohemian Rhapsody`,
      link: `movie-page.html`,
      src: [`1.webm`, `2.mp4`]
    }
  ]
};

it(`renders correctly`, () => {
  const {films} = mock;
  const clickHandler = jest.fn();
  const tree = renderer
    .create(<MovieList
      films={films}
      onClick={clickHandler}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
