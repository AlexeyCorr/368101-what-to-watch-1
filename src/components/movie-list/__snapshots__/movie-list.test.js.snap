// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="player__video"
      muted={true}
      poster="img/bohemian-rhapsody.jpg"
    >
      <source
        src="1.webm"
        type="video/webm"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
</div>
`;
